app/cdap/components/Connections/Browser/GenericBrowser/index.tsx

```diff
@@ -164,10 +164,14 @@ export function GenericBrowser({ selectedConnection }) {
     try {
       const spec = await getPluginSpec(entity, selectedConnection);
       const plugin = spec?.relatedPlugins?.[0];
-
       const properties = plugin?.properties;
       const schema = plugin?.schema;
 
+      // if reference name is already set, don't set default.
+      if (!properties.referenceName) {
+        properties.referenceName = plugin?.name;
```

Comment:
I don't think the change made here is correct, what the JIRA mentioned is to fill in the missing connection name in the connection widget, not the reference name. @njbriggs can you help take a look here?